### PR TITLE
Fix ccds generator from ImageFileCollection (see issue #421)

### PR DESCRIPTION
**Pull request by joshwalawender**
Please have a look at the following list and replace the "[ ]" with a "[x]" if
the answer to this question is yes.

- [ ] For new contributors: Did you add yourself to the "Authors.rst" file?

For documentation changes:

- [ ] For documentation changes: Does your commit message include a "[skip ci]"?
      Note that it should not if you changed any examples!

For bugfixes:

- [x] Did you add an entry to the "Changes.rst" file?
- [ ] Did you add a regression test?
- [x] Does the commit message include a "Fixes #issue_number" (replace "issue_number").
- [ ] Does this PR add, rename, move or remove any existing functions or parameters?

For new functionality:

- [ ] Did you add an entry to the "Changes.rst" file?
- [ ] Did you include a meaningful docstring with Parameters, Returns and Examples?
- [ ] Does the commit message include a "Fixes #issue_number" (replace "issue_number").
- [ ] Did you include tests for the new functionality?
- [ ] Does this PR add, rename, move or remove any existing functions or parameters?

Please note that the last point is not a requirement. It is meant as a check if
the pull request potentially breaks backwards-compatibility.

-----------------------------------------

Fixes issue #421

### CODE REVIEW

**Comment by mirca** [comment]
@joshwalawender nice catch! thank you!

**Comment by MSeifert04** [comment]
LGTM but should be rebased onto current master to resolve the conflict and so the CI can run.

**Comment by bsipocz** [comment]
I wonder whether it would be useful to add a regression test that simulates what @joshwalawender did in #421?

File: CHANGES.rst [hunk 1]
```diff
@@ -22,7 +22,8 @@ Bug Fixes
 
 - ``ccd_process`` now copies the meta of the input when subtracting the
   master bias. [#404]
-
```
Comment (joshwalawender) [inline CHANGES.rst]: Should the number in brackets in CHANGES refer to the Issue #421  or the PR #422 ?

Comment (mwcraig) [inline CHANGES.rst]: It can include both: `[#421, #422]`. If you want to do just one, I'd say do the PR.

**Comment by mwcraig** [comment]
@joshwalawender -- could you please rebase? If you have a chance and can add a test, great, otherwise I'll look at adding one over the weekend.

File: CHANGES.rst (hunk 1 above)
Comment (MSeifert04) [inline CHANGES.rst]: actually I think this PR doesn't need a changelog entry at all because the `location` functionality hasn't been published yet.

**Comment by MSeifert04** [comment]
@joshwalawender I just tried recovering and rebasing your branch and at least with this branch it actually wasn't to hard. I'll post a summary later.

One important thing: Don't do any commits to these branches until they are completly recovered. Otherwise it adds another (complicated) step in between.

**Comment by joshwalawender** [comment]
Thanks @MSeifert04 !  I hadn't gotten around to rebasing this one.  I'm still catching up on various coding tasks.

**Comment by MSeifert04** [comment]
First, I recommend that you copy your branch and test it once with the copy. If any step makes problems or the outcome looks weird one can easily switch the the original branch. Feel free to ask as many questions as you like and if any step makes problems or even if you just want to know what that step does:

## Undo the "master branch merge"

> ``git reset --hard HEAD~1``

## Squash your commits so there is at most one commit that conflicts

Note: The `~2` is because this branch contains only 2 commits for your fix. Adapt this number if the number of commits differs.

This is an interactive step so it will open some files you have to modify (if you use windows don't do this step if you haven't setup your git-editor)

> ``git rebase -i HEAD~2``

- Then it opens a file that looks like this:

```
pick bc36cc7 Fix bug on ccds generator
pick 0f3d5e2 Update listing in CHANGES

# Rebase f964a2c..0f3d5e2 onto f964a2c (2 command(s))
#
# Commands:
# p, pick = use commit
# r, reword = use commit, but edit the commit message
# e, edit = use commit, but stop for amending
# s, squash = use commit, but meld into previous commit
# f, fixup = like "squash", but discard this commit's log message
# x, exec = run command (the rest of the line) using shell
# d, drop = remove commit
#
# These lines can be re-ordered; they are executed from top to bottom.
#
# If you remove a line here THAT COMMIT WILL BE LOST.
#
# However, if you remove everything, the rebase will be aborted.
#
# Note that empty commits are commented out
```

- replace the "pick" in the second line by "s".

- Save changes to the file and close it.

- Another file opens:

```
# This is a combination of 2 commits.
# The first commit's message is:

Fix bug on ccds generator

# This is the 2nd commit message:

Update listing in CHANGES

# Please enter the commit message for your changes. Lines starting
# with '#' will be ignored, and an empty message aborts the commit.
#
# Author:    Josh Walawender <jmwalawender@gmail.com>
# Date:      Wed Nov 16 20:26:43 2016 -1000
#
# interactive rebase in progress; onto f964a2c
# Last commands done (2 commands done):
#    pick bc36cc7 Fix bug on ccds generator
#    s 0f3d5e2 Update listing in CHANGES
# No commands remaining.
# You are currently editing a commit while rebasing branch 'j1' on 'f964a2c'.
#
# Changes to be committed:
#	modified:   CHANGES.rst
#	modified:   ccdproc/image_collection.py
#
```

- Replace the leading lines until (including) the "Update listing in CHANGES" with: "Fix bug on ccds generator"

- Again save and close it.

## Rebase the branch:

You might need to rebase the branch now (this requires that your `master`-branch is up-to-date):

> ``git rebase master <<<your_branch_name>>>``

If it says something about conflicts in Changes.rst then I always do it like this:

- Open the file "CHANGES.rst"

- Open https://raw.githubusercontent.com/astropy/ccdproc/master/CHANGES.rst and copy the contents in your CHANGES.rst

- Include your changelog entry (for example copy it from the changed files screens on github) again.

- Stage the changes (don't commit!!!)

> ``git rebase --continue``

# If rebase tells you there is no conflict

If it tells you in the step "git rebase master <<<your_branch_name>>>" that your current branch is up-to-date it's a lie. You actually need to alter `CHANGES.rst`:

- Copy the https://raw.githubusercontent.com/astropy/ccdproc/master/CHANGES.rst into your CHANGES.rst file and then add your changelog entry again.

- stage and commit the changes.

## Final step:

Push your changes (using -f option for force-push) to your branch.

**Comment by MSeifert04** [comment]
The squash-step is completly optional, but it can make the rebasing easier.

**Comment by joshwalawender** [comment]
Are there any action items on me for this PR?  Trying to catch up on my GitHub tasks and it looks like the rebase has been done (thanks! @MSeifert04 ).

**Comment by bsipocz** [comment]
@joshwalawender - I'm afraid there are still unrelated commits in here, something must have gone wrong during your rebase.

**Comment by MSeifert04** [comment]
> Are there any action items on me for this PR?

Yes, you need to undo the "Merge branch 'fix_collection' of https://github.com/joshwalawender/ccdproc into fix_collection" commit and then rebase the branch on master.

Currently this PR is unmergeable because it would (probably) rewrite the master-branch history if this PR is merged.

**Comment by joshwalawender** [comment]
Let me know if this version is ok.  I think part of the problem stemmed from me accidentally doing work in my local "master" branch for another PR, so I had to fiddle with my local repo a bit.  This level of git-fu is new to me, so I may not have this right.

**Comment by MSeifert04** [comment]
`Changes.rst` picked up some unrelated stuff. Could you revert the changes there except for your entry?

I'm really sorry for the long delay on this PR.

**Comment by mwcraig** [comment]
@joshwalawender -- thanks for the changes here! This needed another rebase; I've opened #431, which contains these changes and fixes the rebase issue. Assuming those tests pass, your changes will be in version 1.2, released later this week.

Closing this PR.

**Comment by MSeifert04** [comment]
#431 is merged now, thanks @joshwalawender 👍

**Pull request closed without merging**